### ISSUE TITLE
Storing and Retrieving Map objects is changing datatype to array

### ISSUE THREAD

**Issue by jbuddha** [championswimmer/vuex-persist#112]
I'm trying to store a state that contains a map property.  

```
 state: {
    user: null,
    ordered_items: new Map()
}

```
Mutations and everything are happening fine, where I add some key value pairs to ordered items map, but when I reload the page. Ordered items is coming back as an Array of Array. Which causes issues in mutation function that is like this... It says method set can't be found. Is there any workaround for this? 

```
[ADD_ITEM]: (state, id, order_time) =>
      state.ordered_items.set([id, order_time]),
```

**Comment by tmladek** [comment]
Yes, you need to only use JSON serializable objects (which `Map` unfortunately is not).

A workaround might be to retrieve and store an Array, but convert it into and from a Map out of it when necessary (for example in mutations and getters).
